### PR TITLE
Verify receive address

### PR DESCRIPTION
**Pull request by edouardparis**

### CODE REVIEW

**Comment by darosior** [comment]
A rebase on master would fix the CI, i think. (See #824.)

**Comment by darosior** [comment]
I don't know if it has anything to do with the implementation here, but after verifying the address on my Ledger Nano S (not plus) my Ledger's screen stay stuck on "Approve". Trying again unstucks it but it gets stuck again after i approved it.

**Comment by darosior** [comment]
If i trigger an error (reject the verification or otherwise disconnect the Ledger) it doesn't get cleaned up after i try again and approve the address.
image

**Comment by edouardparis** [comment]
rebased on master to satisfy the gui.

File: gui/src/app/view/receive.rs [hunk 1]
```diff
@@ -115,3 +132,53 @@ pub fn receive<'a>(
...
+        .push_maybe(warning.map(|w| warn(Some(w))))
+        .push(card::simple(
+            Column::new()
+                .push(
+                    Column::new()
+                        .push(
+                            Container::new(
+                                Row::new()
+                                    .push(text(address.to_string()).bold())
+                                    .push(text(format!("#{}", derivation_index)))
```
Comment (darosior) [inline gui/src/app/view/receive.rs:153]: It's not exactly clear this refers to. What do you think of changing the display to:
- Address: `b1rt....`
- Derivation index: `XXX`

**Comment by darosior** [comment]
This looks great! :rocket:

**Comment by edouardparis** [comment]
fixed the two comments and rebase, I do not know for the Ledger Nano.

**Comment by Rob1Ham** [comment]
> I don't know if it has anything to do with the implementation here, but after verifying the address on my Ledger Nano S (not plus) my Ledger's screen stay stuck on "Approve". Trying again unstucks it but it gets stuck again after i approved it. Same with "Reject".

For what it is worth, I've seen this issue as well using the javascript library as well.

**Comment by darosior** [comment]
cc @bigspider: the above two comments may be of interest to you.

**Comment by darosior** [comment]
Tested this on a mainnet wallet with a Ledger Nano S. Looks good to me. Will now ask @jp1ac4 to have a look at the code.

**Comment by bigspider** [comment]
> cc @bigspider: the above two comments may be of interest to you.

Please let me know how I can try to reproduce (e.g. policy and PSBT you were signing) and I'll give this a look

**Review by jp1ac4 — APPROVED** [review]
utACK https://github.com/wizardsardine/liana/pull/819/commits/e9a217b3d87fc5ccd284df33fc58f877c7a5103e.

As well as reviewing the code, I checked the receive screen, but did not verify any addresses on a device.

**Comment by darosior** [comment]
@edouardparis i trust you tested this with Bitbox and the Nano S+?

**Pull request merged**